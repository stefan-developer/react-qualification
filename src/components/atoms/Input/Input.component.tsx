import React from "react";
import * as Styled from "./Input.styles";

interface InputProps {
  onChange: (value: string) => void;
  value: string;
  placeholder: string;
}

function Input(props: InputProps): JSX.Element {
  const { onChange, value, placeholder } = props;

  return (
    <Styled.Wrapper>
      <input onChange={(event) => onChange(event.target.value)} value={value} placeholder={placeholder}/>
    </Styled.Wrapper>
  );
}

export default Input as React.FC<InputProps>;
