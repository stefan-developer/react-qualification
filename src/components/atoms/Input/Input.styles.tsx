import styled from "styled-components";

export const Wrapper = styled.div`
  width: 100%;
  max-width: 600px;
  > input {
    font-family: "Open Sans", sans-serif;
    width: 100%;
    padding: 10px;
  }
`;
