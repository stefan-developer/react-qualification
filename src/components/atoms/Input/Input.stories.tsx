import React from "react";

import { storiesOf } from "@storybook/react";

import Input from "./Input.component";

storiesOf("Input", module).add("default", () => (
  <Input placeholder="Search" onChange={() => null} value="hello" />
));
