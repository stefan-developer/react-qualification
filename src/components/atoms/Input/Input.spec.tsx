import React from "react";
import renderer from "react-test-renderer";

import Input from "./Input.component";

test("Input renders as expected", () => {
    const component = renderer.create(
        <Input placeholder="Test example" onChange={() => null} value="Test value" />
    );

    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
});