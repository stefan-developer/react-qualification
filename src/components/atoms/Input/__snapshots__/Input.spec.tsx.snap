// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input renders as expected 1`] = `
<div
  className="sc-AxjAm cRFgOx"
>
  <input
    onChange={[Function]}
    placeholder="Test example"
    value="Test value"
  />
</div>
`;
